### PR TITLE
Adds strong_migrations gem. Closes #214

### LINKED ISSUE #214: Consider adding https://github.com/ankane/strong_migrations to automated checks
[strong_migrations](https://github.com/ankane/strong_migrations) runs checks on pending migrations to check that they are safe to run. 'Safe' in this context means that they should not unexpectedly block table reads or writes, or potentially lose data. 

This is probably a good idea for us to add to our automated checks - understanding the impact of migrations often requires some knowledge of both the underlying SQL statements _and_ the drawbacks/capabilities of the RDMS in use. This check would help less experienced developers understand the impact of a migration, and any potential optimisations to help the migration to run in a safer way.

### PR DESCRIPTION
**Pull request by dp4rk**
Have tested this different variants, and all of the existing default migrations pass without errors.

### CODE REVIEW

File: bin/setup
Comment (dp4rk) [inline bin/setup:14]: I've done it this way because the config file that it creates automatically pulls in the time that it's run. Theoretically could just copy in the initializer that this generates, but this way is more dynamic.

**Comment by dp4rk** [comment]
This is on hold - it is a "nice to have" but it is preventative for early project work because we need to be able to do unsafe migrations early on and this gem forces errors.

If we can configure this gem to raise warnings instead of errors then we definitely are good with this, but right now this is not practical. 

We will want to potentially implement this gem on finished projects but not at the beginning stage.

**Comment by eoinkelly** [comment]
I'm going to close this because we don't have a plan to implement this anytime soon. Thank you for the contribution and the surrounding discussion @dp4rk

**Comment by eoinkelly** [comment]
This is an example of the initializer it generates - we probably want to discuss the details here:
```ruby
# Mark existing migrations as safe
StrongMigrations.start_after = 20_231_006_001_223

# Set timeouts for migrations
# If you use PgBouncer in transaction mode, delete these lines and set timeouts on the database user
StrongMigrations.lock_timeout = 10.seconds
StrongMigrations.statement_timeout = 1.hour

# Analyze tables after indexes are added
# Outdated statistics can sometimes hurt performance
StrongMigrations.auto_analyze = true

# Set the version of the production database
# so the right checks are run in development
# StrongMigrations.target_version = 10

# Add custom checks
# StrongMigrations.add_check do |method, args|
#   if method == :add_index && args[0].to_s == "users"
#     stop! "No more indexes on the users table"
#   end
# end

# Make some operations safe by default
# See https://github.com/ankane/strong_migrations#safe-by-default
# StrongMigrations.safe_by_default = true
```

**Comment by eoinkelly** [comment]
CI is failing until PR #502 lands

**Review by JacobBriggsAckama — APPROVED** [review]

**Comment by eoinkelly** [comment]
Closes #214

**Pull request merged**
